### PR TITLE
Add support for time type in Cassandra

### LINKED ISSUE #13063: Cassandra Connector support for CQL TIME data type
Currently the Cassandra connector does not support the Cassandra TIME data type in columns, primary keys or in 'CLUSTERING ORDER BY' clause.  

Example CQL:
```
CREATE TABLE test.datausagetable (
    dbname text,
    device_id text,
    vd text,
    tstamp_date date,
    tstamp_time time,
    count_file_bytes int,
    count_files int,
    count_logs bigint,
    count_tcp_bytes int,
    count_udp_bytes int,
    PRIMARY KEY ((dbname, device_id, vd), tstamp_date, tstamp_time)
) WITH CLUSTERING ORDER BY (tstamp_date DESC, tstamp_time DESC) ....;
```

REFERENCE: https://cassandra.apache.org/doc/latest/cassandra/cql/types.html

Values of the TIME type are encoded as 64-bit signed integers representing the number of nanoseconds since midnight.

Our minimum requirements would be for Cassandra 3+ and going foward.
Currently using: [cqlsh 5.0.1 | Cassandra 3.11.13 | CQL spec 3.4.4 | Native protocol v4]

Current Trino CLI-387 output:

```
DESCRIBE datausagetable;
      Column      |            Type             | Extra | Comment 
------------------+-----------------------------+-------+---------
 dbname           | varchar                     |       |         
 device_id        | varchar                     |       |         
 vd               | varchar                     |       |         
 tstamp_date      | date                        |       |         
 count_file_bytes | integer                     |       |         
 count_files      | integer                     |       |         
 count_logs       | bigint                      |       |         
 count_tcp_bytes  | integer                     |       |         
 count_udp_bytes  | integer                     |       |         
 secconlogger     | timestamp(3) with time zone |       |         
```
(10 rows)

Query 20220701_141009_00450_4w2mq, FINISHED, 3 nodes
Splits: 53 total, 53 done (100.00%)
0.14 [10 rows, 783B] [71 rows/s, 5.46KB/s]

`EXPLAIN SELECT * FROM datausagetable WHERE dbname='FG_NAME' AND device_id='FG_ID' AND vd='' AND tstamp_date <= cast( '2022-01-16' as date);`
```
                                                                                                                     Query Plan                                                                                    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 0 [SINGLE]                                                                                                                                                                                               
     Output layout: [dbname, device_id, vd, tstamp_date, count_file_bytes, count_files, count_logs, count_tcp_bytes, count_udp_bytes, secconlogger]                                                                
     Output partitioning: SINGLE []                                                                                                                                                                                
     Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                                                                                                 
     Output[dbname, device_id, vd, tstamp_date, count_file_bytes, count_files, count_logs, count_tcp_bytes, count_udp_bytes, secconlogger]                                                                         
     │   Layout: [dbname:varchar, device_id:varchar, vd:varchar, tstamp_date:date, count_file_bytes:integer, count_files:integer, count_logs:bigint, count_tcp_bytes:integer, count_udp_bytes:integer, secconlogger
     │   Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: ?}                                                                                                                                                  
     └─ RemoteSource[1]                                                                                                                                                                                            
            Layout: [dbname:varchar, device_id:varchar, vd:varchar, tstamp_date:date, count_file_bytes:integer, count_files:integer, count_logs:bigint, count_tcp_bytes:integer, count_udp_bytes:integer, secconlog
                                                                                                                                                                                                                   
 Fragment 1 [SOURCE]                                                                                                                                                                                               
     Output layout: [dbname, device_id, vd, tstamp_date, count_file_bytes, count_files, count_logs, count_tcp_bytes, count_udp_bytes, secconlogger]                                                                
     Output partitioning: SINGLE []                                                                                                                                                                                
     Stage Execution Strategy: UNGROUPED_EXECUTION                                                                                                                                                                 
     TableScan[cassandra:nusiem:datausagetable, grouped = false]                                                                                                                                                   
         Layout: [dbname:varchar, device_id:varchar, vd:varchar, tstamp_date:date, count_file_bytes:integer, count_files:integer, count_logs:bigint, count_tcp_bytes:integer, count_udp_bytes:integer, secconlogger
         Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}                                                                                                                                                 
         count_udp_bytes := CassandraColumnHandle{name=count_udp_bytes, ordinalPosition=9, cassandraType=INT, partitionKey=false, clusteringKey=false, indexed=false, hidden=false}                                
         dbname := CassandraColumnHandle{name=dbname, ordinalPosition=0, cassandraType=TEXT, partitionKey=true, clusteringKey=false, indexed=false, hidden=false}                                                  
         device_id := CassandraColumnHandle{name=device_id, ordinalPosition=1, cassandraType=TEXT, partitionKey=true, clusteringKey=false, indexed=false, hidden=false}                                            
         count_tcp_bytes := CassandraColumnHandle{name=count_tcp_bytes, ordinalPosition=8, cassandraType=INT, partitionKey=false, clusteringKey=false, indexed=false, hidden=false}                                
         count_logs := CassandraColumnHandle{name=count_logs, ordinalPosition=7, cassandraType=BIGINT, partitionKey=false, clusteringKey=false, indexed=false, hidden=false}                                       
         count_files := CassandraColumnHandle{name=count_files, ordinalPosition=6, cassandraType=INT, partitionKey=false, clusteringKey=false, indexed=false, hidden=false}                                        
         tstamp_date := CassandraColumnHandle{name=tstamp_date, ordinalPosition=3, cassandraType=DATE, partitionKey=false, clusteringKey=true, indexed=false, hidden=false}                                        
         secconlogger := CassandraColumnHandle{name=secconlogger, ordinalPosition=10, cassandraType=TIMESTAMP, partitionKey=false, clusteringKey=false, indexed=false, hidden=false}                               
         vd := CassandraColumnHandle{name=vd, ordinalPosition=2, cassandraType=TEXT, partitionKey=true, clusteringKey=false, indexed=false, hidden=false}                                                          
         count_file_bytes := CassandraColumnHandle{name=count_file_bytes, ordinalPosition=5, cassandraType=INT, partitionKey=false, clusteringKey=false, indexed=false, hidden=false}                              

```                                                                                                                                                                                                                   
This is a high priority for my company so as a Senior Java Developer I will do all I can to assist and/or contribute to get this implemented.  Started looking at the CassandraTypes class per advice from Trino developer.  
Of course, any questions, comments or suggestions are welcome.

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #13063

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Cassandra
* Add support for `time` type. ({issue}`13063`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
`TestCassandraConnectorTest.testUserDefinedTypeInSet` will be fixed in #15805

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTypeMapping.java [hunk 1]
```diff
@@ -507,6 +509,39 @@ private SqlDataTypeTest dateTest(Function<String, String> inputLiteralFactory)
...
+                .execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_time"))
+                .execute(getQueryRunner(), session, trinoCreateAndInsert("test_time"));
+
+        timeTypeTest("time", cassandraTimeInputLiteralFactory())
+                .execute(getQueryRunner(), session, cassandraCreateAndInsert("tpch.test_time"));
+    }
+
+    private static SqlDataTypeTest timeTypeTest(String inputType, Function<String, String> inputLiteralFactory)
+    {
+        return SqlDataTypeTest.create()
```
Comment (hashhar) [inline plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTypeMapping.java:534]: nulls?

**Review by hashhar — COMMENTED** [review]
LGTM % test for nulls

File: plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTypeMapping.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTypeMapping.java:534]: Added another roundtrip.

**Review by hashhar — APPROVED** [review]

**Pull request merged**